src/services/heartbeat.ts

```diff
@@ -0,0 +1,34 @@
+import EventEmitter from "node:events";
+import { env } from "../utils/env";
+import { defaultEmitter } from "../utils/emitter";
+import { BotEventNames } from "../bot-events";
+
+export let interval: NodeJS.Timeout | undefined;
+
+export const initialize = () => configurableInitialize({ env, emitter: defaultEmitter });
+
+type ConfigurableInitializeDeps = {
+  env: typeof env;
+  emitter: EventEmitter;
+};
+export const configurableInitialize = (deps: ConfigurableInitializeDeps) => {
+  const { env, emitter } = deps;
+
+  const { HEARTBEAT_URL, HEARTBEAT_INTERVAL } = env;
+  if (!HEARTBEAT_URL || !HEARTBEAT_INTERVAL) {
+    emitter.emit(BotEventNames.HEARTBEAT_CONFIGURATION_MISSING);
+    return;
+  }
+
+  if (interval) clearInterval(interval);
```

Comment:
When does this line come into play?

Reply:
@wadader If the service has been already configured and by mistake it is called twice or more times, we remove any previous periodic request (modeled with intervals) before setting the new one. This helps avoiding floods while preserving the ability to call the function multiple times in tests / reconfigure it.